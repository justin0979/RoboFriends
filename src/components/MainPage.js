import React from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundary';
import Header from '../components/Header';
import './MainPage.scss';

class MainPage extends React.Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter(robot =>
      robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
    );
  };

  pendingStatus = () => {
    return this.props.isPending ? (
      <h1>Loading</h1>
    ) : (
      <ErrorBoundary>
        <CardList robots={this.filterRobots()} />
      </ErrorBoundary>
    );
  };

  render() {
    if (this.props.isPending) return <h1>Loading</h1>;
    return (
      <div className="tc">
        <Header />
        <SearchBox onChange={this.props.handleChange} />
        <Scroll>{this.pendingStatus()}</Scroll>
      </div>
    );
  }
}

export default MainPage;
