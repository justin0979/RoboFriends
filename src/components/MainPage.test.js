import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it('should render the MainPage without crashing', () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

it('should filter robots correctly', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: 'Jon',
        email: 'jon@gmail.com'
      }
    ],
    searchField: 'jon',
    isPending: false
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots()).toEqual(mockProps2.robots);
});

it('should filter robots correctly', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: 'Jon',
        email: 'jon@gmail.com'
      }
    ],
    searchField: 'a',
    isPending: false
  };
  const filteredRobots = [];
  const wrapper3 = shallow(<MainPage {...mockProps3} />);
  expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
});

it('should filter robots correctly', () => {
  const mockProps4 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: 'Jon',
        email: 'jon@gmail.com'
      }
    ],
    searchField: 'a',
    isPending: true
  };
  const filteredRobots = [];
  const wrapper4 = shallow(<MainPage {...mockProps4} />);
  expect(wrapper4.instance().pendingStatus()).toEqual(<h1>Loading</h1>);
});
