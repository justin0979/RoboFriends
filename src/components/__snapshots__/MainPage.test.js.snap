// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the MainPage without crashing 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox onChange={[undefined]} />
  <Scroll>
    <ErrorBoundary>
      <CardList robots={{...}} />
    </ErrorBoundary>
  </Scroll>
</div>"
`;
